<a href="https://colab.research.google.com/github/kanokwan-hongsopa/PythonDataViz2026/blob/main/HW7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**HW7**

- สร้างตารางใหม่ ที่มีค่าใน sex เป็น missing ทั้งหมด
- สรุปว่าทำไม record นั้นๆ ถึงเป็น missing (ไม่ใช้ groupby)

In [1]:
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
import os

path = '/content/drive/My Drive/PythonDataViz2026'

In [6]:
covid_file_path = os.path.join(path,'confirmed-cases.csv')
print(covid_file_path)

/content/drive/My Drive/PythonDataViz2026/confirmed-cases.csv


In [7]:
a = path + '/confirmed-cases.csv'
print(a)

/content/drive/My Drive/PythonDataViz2026/confirmed-cases.csv


In [8]:
data_covid = pd.read_csv(covid_file_path)
data_covid

/tmp/ipykernel_1001/318669438.py:1: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  data_covid = pd.read_csv(covid_file_path)


,No.,announce_date,Notified date,sex,age,Unit,nationality,province_of_isolation,risk,province_of_onset,district_of_onset
0,1,12/1/2020,NaN,หญิง,61.0,ปี,China,กรุงเทพมหานคร,คนต่างชาติเดินทางมาจากต่างประเทศ,กรุงเทพมหานคร,NaN
1,2,17/1/2020,NaN,หญิง,74.0,ปี,China,กรุงเทพมหานคร,คนต่างชาติเดินทางมาจากต่างประเทศ,กรุงเทพมหานคร,NaN
2,3,22/1/2020,NaN,หญิง,73.0,ปี,Thailand,นครปฐม,คนต่างชาติเดินทางมาจากต่างประเทศ,นครปฐม,เมือง
3,4,22/1/2020,NaN,ชาย,68.0,ปี,China,กรุงเทพมหานคร,คนต่างชาติเดินทางมาจากต่างประเทศ,กรุงเทพมหานคร,NaN
4,5,24/1/2020,NaN,หญิง,66.0,ปี,China,นนทบุรี,คนต่างชาติเดินทางมาจากต่างประเทศ,กรุงเทพมหานคร,NaN
...,...,...,...,...,...,...,...,...,...,...,...
839766,839767,12/8/2021,11/8/2021,ชาย,52.0,ปี,NaN,กาญจนบุรี,ทัณฑสถาน/เรือนจำ,กาญจนบุรี,NaN
839767,839768,12/8/2021,11/8/2021,ชาย,34.0,ปี,Thailand,กาญจนบุรี,ทัณฑสถาน/เรือนจำ,กาญจนบุรี,NaN
839768,839769,12/8/2021,11/8/2021,ชาย,31.0,ปี,Thailand,กาญจนบุรี,ทัณฑสถาน/เรือนจำ,กาญจนบุรี,NaN
839769,839770,12/8/2021,11/8/2021,ชาย,83.0,ปี,Thailand,กาญจนบุรี,ทัณฑสถาน/เรือนจำ,กาญจนบุรี,NaN


In [14]:
data_covid.isnull()

,No.,announce_date,Notified date,sex,age,Unit,nationality,province_of_isolation,risk,province_of_onset,district_of_onset
0,False,False,True,False,False,False,False,False,False,False,True
1,False,False,True,False,False,False,False,False,False,False,True
2,False,False,True,False,False,False,False,False,False,False,False
3,False,False,True,False,False,False,False,False,False,False,True
4,False,False,True,False,False,False,False,False,False,False,True
...,...,...,...,...,...,...,...,...,...,...,...
839766,False,False,False,False,False,False,True,False,False,False,True
839767,False,False,False,False,False,False,False,False,False,False,True
839768,False,False,False,False,False,False,False,False,False,False,True
839769,False,False,False,False,False,False,False,False,False,False,True


In [17]:
data_covid.isnull().any() # falseไม่มีข้อมูลสูญหาย
                          # True มีข้อมูลหายอย่างน้อย 1 ค่า

,0
No.,False
announce_date,False
Notified date,True
sex,True
age,True
Unit,True
nationality,True
province_of_isolation,True
risk,True
province_of_onset,True


In [19]:
missing_sex = data_covid[data_covid['sex'].isnull()]

In [16]:
print(missing_sex)

           No. announce_date Notified date  sex   age Unit nationality  \
4397      4398    20/12/2020           NaN  NaN   NaN  NaN       Burma   
4398      4399    20/12/2020           NaN  NaN   NaN  NaN       Burma   
4399      4400    20/12/2020           NaN  NaN   NaN  NaN       Burma   
4400      4401    20/12/2020           NaN  NaN   NaN  NaN       Burma   
4401      4402    20/12/2020           NaN  NaN   NaN  NaN       Burma   
...        ...           ...           ...  ...   ...  ...         ...   
838332  838333     12/8/2021     11/8/2021  NaN   3.0   ปี    Thailand   
839174  839175     12/8/2021     11/8/2021  NaN  47.0   ปี    Thailand   
839304  839305     12/8/2021     11/8/2021  NaN   NaN  NaN         NaN   
839396  839397     12/8/2021     11/8/2021  NaN   NaN  NaN         NaN   
839397  839398     12/8/2021     11/8/2021  NaN   NaN  NaN         NaN   

       province_of_isolation                                         risk  \
4397               สมุทรสาคร      

In [22]:
missing_sex.shape[0]

54895

สรุป
- จากการวิเคราะห์พบว่ามีผู้ป่วยที่ไม่ทราบเพศถึง 54895 ราย
- สาเหตุที่เป็นไปได้ของการเกิด missing ที่ recordนั้นๆ :

สาเหตุที่เป็นไปได้ เช่น

    * ผู้ป่วยไม่ได้แจ้งข้อมูล
    * เจ้าหน้าที่ไม่ได้บันทึกข้อมูล
    * ข้อมูลยังเก็บไม่ครบ
    * ข้อมูลสูญหายระหว่างการรวบรวม

In [21]:
print(missing_sex.isnull().sum())

No.                          0
announce_date                0
Notified date             1541
sex                      54895
age                      25403
Unit                     25259
nationality              38085
province_of_isolation        0
risk                        79
province_of_onset        20567
district_of_onset        24924
dtype: int64
